src/core/lib.lua

```diff
@@ -386,12 +353,64 @@ if ffi.abi("be") then
    function htonl(b) return b end
    function htons(b) return b end
 else
-   function htonl(b) return bit.bswap(b) end
-   function htons(b) return bit.rshift(bit.bswap(b), 16) end
+   function htonl(b) return bswap(b) end
+   function htons(b) return rshift(bswap(b), 16) end
 end
 ntohl = htonl
 ntohs = htons
 
+-- The fact that BitOps return signed integers is irritating, to say
+-- the least.  One counter-intuitive consequence is that, for example,
+--
+--   bit.bswap(bit.bswap(0x80000000)) == 0x80000000
```

Comment:
Ok, thanks. Maybe one day I'll even understand why BitOps using signed integers is considered a good thing in the first place :)

I'll remove this function and call bit.tobit explicitely when it's needed.